pkg/gatewayserver/io/io.go

```diff
@@ -214,6 +214,8 @@ func (c *Connection) HandleUp(up *ttnpb.UplinkMessage) error {
 			if location.Source != ttnpb.SOURCE_UNKNOWN {
```

Comment:
When does this happen? Shouldn't the location be set from the registry if and only if the source is registry?

Reply:
When the location is updated automatically from status messages https://github.com/TheThingsNetwork/lorawan-stack/blob/db69613dec92c40815d2596709b4b5f3c3f6ea62/pkg/gatewayserver/gatewayserver.go#L791-L792